sdk/attestation/azure-security-attestation/azure/security/attestation/_administration_client.py

```diff
@@ -4,316 +4,736 @@
+    :keyword float validation_slack: Slack time for validation - tolerance applied
+        to help account for clock drift between the issuer and the current machine.
+    :keyword bool validate_issuer: If True, validate that the issuer of the token matches the expected issuer.
+    :keyword bool validate_not_before_time: If true, validate the "Not Before" time in the token.
+
+    If the `signing_key` and `signing_certificate` parameters
+    are provided, they will be applied to the following APIs:
+
+    * :py:func:`set_policy`
+    * :py:func:`reset_policy`
+    * :py:func:`add_policy_management_certificate`
+    * :py:func:`remove_policy_management_certificate`
+
+    .. note::
+        The `signing_key` and `signing_certificate` parameters are a pair. If one
+        is present, the other must also be provided. In addition, the public key
+        in the `signing_key` and the public key in the `signing_certificate` must
+        match to ensure that the `signing_certificate` can be used to validate an
+        object signed by `signing_key`.
+
+    .. tip::
+        The `validate_token`, `validation_callback`, `validate_signature`,
+        `validate_expiration`, `validate_not_before_time`, `validate_issuer`, and
+        `issuer` keyword arguments are default values applied to each API call within
+        the :py:class:`AttestationAdministrationClient` class. These values can be
+        overridden on individual API calls as needed.
     """
 
     def __init__(
         self,
+        endpoint,  # type: str
         credential,  # type: "TokenCredential"
-        instance_url,  # type: str
         **kwargs  # type: Any
     ):
         # type: (...) -> None
         if not credential:
             raise ValueError("Missing credential.")
-        self._config = AttestationClientConfiguration(credential, instance_url, **kwargs)
-        self._client = AzureAttestationRestClient(credential, instance_url, **kwargs)
+        self._config = AttestationClientConfiguration(**kwargs)
+        self._client = AzureAttestationRestClient(credential, endpoint, **kwargs)
         self._statelock = Lock()
         self._signing_certificates = None
 
+        self._signing_key = None
+        self._signing_certificate = None
+
+        signing_key = kwargs.pop("signing_key", None)
+        signing_certificate = kwargs.pop("signing_certificate", None)
+        if signing_key or signing_certificate:
+            self._signing_key, self._signing_certificate = validate_signing_keys(
+                signing_key, signing_certificate
+            )
+
     @distributed_trace
-    def get_policy(self, attestation_type, **kwargs): 
-        #type(AttestationType, **Any) -> AttestationResponse[str]:
-        """ Retrieves the attestation policy for a specified attestation type.
+    def get_policy(self, attestation_type, **kwargs):
+        # type: (Union[str, AttestationType], **Any) -> Tuple[str, AttestationToken]
+        # pylint: disable = line-too-long
```

Comment:
Since a `line-too-long` suppression is present at the start of this class, you don't have to suppress the warning here or elsewhere in the method (I think there's currently a suppression at the start of each method)
```suggestion
```

Reply:
Yeah - I was trying to do it on a case-by-case basis, but the line-too-long also hit the class docstr, so I was forced to put it at the top of the class. Annoying, but it is what it is.